mopidy/files/library.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import unicode_literals
+
+import logging
+import operator
+import os
+import sys
+import urllib2
+
+from mopidy import backend, exceptions, models
+from mopidy.audio import scan, utils
+from mopidy.internal import path
+
+logger = logging.getLogger(__name__)
+
+
+class FilesLibraryProvider(backend.LibraryProvider):
+    """Library for browsing local files."""
+    # TODO: get_images that can pull from metadata and/or .folder.png etc?
+    # TODO: handle playlists?
+
+    @property
+    def root_directory(self):
+        if not self._media_dirs:
+            return None
+        elif len(self._media_dirs) == 1:
+            local_path = self._media_dirs[0]['path']
+            uri = path.path_to_uri(local_path)
+        else:
+            uri = 'file:root'
+        return models.Ref.directory(name='Files', uri=uri)
+
+    def __init__(self, backend, config):
+        super(FilesLibraryProvider, self).__init__(backend)
+        self._media_dirs = list(self._get_media_dirs(config))
+        self._follow_symlinks = config['files']['follow_symlinks']
+        self._show_dotfiles = config['files']['show_dotfiles']
+        self._scanner = scan.Scanner(
+            timeout=config['files']['metadata_timeout'])
+
+    def browse(self, uri):
+        logger.debug('Browsing files at: %s', uri)
+        result = []
+        local_path = path.uri_to_path(uri)
+        if local_path == 'root':
+            return list(self._get_media_dirs_refs())
+        for dir_entry in os.listdir(local_path):
+            child_path = os.path.join(local_path, dir_entry)
+            uri = path.path_to_uri(child_path)
+            printable_path = child_path.decode(sys.getfilesystemencoding(),
```

Comment:
It's tempting to replace all the `sys.getfilesystemencoding()` calls with an `FS_ENCODING` constant defined at top level in this file.